mfaliquot/application/updater.py

```diff
@@ -158,14 +158,26 @@ def check_update(self, old):
           if not old or not old.is_minimally_valid() or not old.id:
                return self.do_update(old)
 
-          status = self._fdb_error_handler_wrapper(fdb.query_id_status, old.seq, old.id)
+          status, factors, cofactor = self._fdb_error_handler_wrapper(fdb.query_id_status, old.seq, old.id)
           if not status: # the wrapper has logged it and set self.quitting as necessary
                return old, False
 
           if status is fdb.FDBStatus.CompositeFullyFactored:
                return self.do_update(old)
-          elif status is fdb.FDBStatus.CompositePartiallyFactored: # no progress since last
-               old.process_no_progress()
+          elif status is fdb.FDBStatus.CompositePartiallyFactored:
+               if old.factors == factors:
+                    # no progress since last
+                    old.process_no_progress()
+               else:
+                    # a new cofactor was found on the same index, copy old to update sequence
+                    ali = AliquotSequence(seq=old.seq, size=old.size, index=old.index, id=old.id)
```

Comment:
I'm not really sure why you go to the bother of creating a whole new AliquotSequence object? The old one is still perfectly valid, just needs the new `factors` information copied to it